packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -700,6 +702,18 @@ export default function(
     instance.context = getMaskedContext(workInProgress, unmaskedContext);
 
     if (__DEV__) {
+      const componentName = getComponentName(workInProgress) || 'Component';
+      if (!didWarnAboutDirectlyAssigningPropsToState.has(componentName)) {
+        didWarnAboutDirectlyAssigningPropsToState.add(componentName);
+        warning(
+          instance.state !== instance.props,
```

Comment:
I think we should move this check up a level:
```js
if (__DEV__) {
  if (instance.state === instance.props) {
    const componentName = getComponentName(workInProgress) || 'Component';
    if (!didWarnAboutDirectlyAssigningPropsToState.has(componentName)) {
      didWarnAboutDirectlyAssigningPropsToState.add(componentName);
      warning(false, ...);
```
This way we don't add things to the Set unnecessarily, and we also don't potentially suppress a later warning (e.g. if we encounter 2 components with the same name- and the second one assigns props to state).

Reply:
Ah, yes, of course. Thanks for catching that.